libs/wingsdk/src/target-sim/schedule.inflight.ts

```diff
@@ -0,0 +1,81 @@
+import { IFunctionClient, IScheduleClient, TraceType } from "../cloud";
+import { ISimulatorContext, ISimulatorResourceInstance } from "../testing";
+import { IEventPublisher } from "./event-mapping";
+import { EventSubscription, SCHEDULE_TYPE, ScheduleAttributes, ScheduleSchema, ScheduleTask } from "./schema-resources";
+import { CronExpression, parseExpression } from "cron-parser";
+
+export class Schedule implements IScheduleClient, ISimulatorResourceInstance, IEventPublisher {
+  private readonly context: ISimulatorContext;
+  private tasks = new Array<ScheduleTask>();
+  private interval: CronExpression;
+  private intervalTimeout?: NodeJS.Timeout;
+
+  constructor(props: ScheduleSchema["props"], context: ISimulatorContext) {
+    this.context = context;
+    this.interval = parseExpression(props.cronExpression);
+    this.scheduleFunction();
+  }
+
+  // Calculate the delay for the next execution
+  private async nextDelay(interval: CronExpression) {
+    return interval.next().toDate().getTime() - Date.now();
+  }
+
+  // Recursively schedule the function to be executed
+  private async scheduleFunction() {
```

Comment:
Can these functions be sync?

Reply:
yea the sure can.. 🤔 why did I make them async to begin with must have been a lapse in judgment 😅